The examples you've shown seem to all have the binding generation part of their buildsystem. That's not the case for libnl so I will still need some kind of python packaging (to run python tests)
I ended up moving the `python` folder to a `py` output but it just contains the source, not a python package.
Also there seems to be tests to run too. I managed to 
```
{ buildPythonPackage, pkgconfig, src, isPy3k, libnl, swig, file }:
buildPythonPackage {
  pname = "libnl";
  inherit (libnl) version  patches meta;

  src=libnl.py;

  disabled = isPy3k;

  buildInputs = [ pkgconfig swig libnl libnl.dev ];
}
```
what I don't get though is why the python package won't compile even though I provided the  libnl.dev output aong with pkgconfig; the correct include path (sthg like /nix/store/d29dwbldfg8qsgcsgp4ic14av9iw5rlw-libnl-3.4.0-dev/include/libnl3) won't appear in gcc command line:
```

building 'netlink/_capi' extension
swigging ./netlink/capi.i to ./netlink/capi_wrap.c
swig -python -O -nodefaultctor -o ./netlink/capi_wrap.c ./netlink/capi.i
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/netlink
gcc -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I../include -I../include -I/nix/store/l9j2jsc9flrbmpf799nw9wdq59gpwms8-python-2.7.14/include/python2.7 -c ./netlink/capi_wrap.c -o build/temp.linux-x86_64-2.7/./netlink/capi_wrap.o
./netlink/capi_wrap.c:3070:10: fatal error: netlink/netlink.h: No such file or directory
 #include <netlink/netlink.h>
          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
error: command 'gcc' failed with exit status 1
note: keeping build directory '/tmp/nix-build-python2.7-libnl-3.4.0.drv-0'

```

--- a/pkgs/os-specific/linux/libnl/default.nix
+++ b/pkgs/os-specific/linux/libnl/default.nix
@@ -1,14 +1,14 @@
 { stdenv, lib, fetchFromGitHub, fetchpatch, autoreconfHook, bison, flex, pkgconfig }:
 
-let version = "3.3.0"; in
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   name = "libnl-${version}";
+  version = "3.4.0";
 
   src = fetchFromGitHub {
     repo = "libnl";
     owner = "thom311";
     rev = "libnl${lib.replaceStrings ["."] ["_"] version}";
-    sha256 = "1796kyq2lkhz2802v9kp32vlxf8ynlyqgyw9nhmry3qh5d0ahcsv";
+    sha256 = "1bqf1f5glwf285sa98k5pkj9gg79lliixk1jk85j63v5510fbagp";
   };
 
   outputs = [ "bin" "dev" "out" "man" ];

--- a/pkgs/os-specific/linux/libnl/default.nix
+++ b/pkgs/os-specific/linux/libnl/default.nix
@@ -1,4 +1,5 @@
-{ stdenv, lib, fetchFromGitHub, fetchpatch, autoreconfHook, bison, flex, pkgconfig }:
+{ stdenv, file, lib, fetchFromGitHub, fetchpatch, autoreconfHook, bison, flex, pkgconfig
+, pythonSupport ? true, swig ? null, python}:
 
 stdenv.mkDerivation rec {
   name = "libnl-${version}";
@@ -11,7 +12,7 @@ stdenv.mkDerivation rec {
     sha256 = "1bqf1f5glwf285sa98k5pkj9gg79lliixk1jk85j63v5510fbagp";
   };
 
-  outputs = [ "bin" "dev" "out" "man" ];
+  outputs = [ "bin" "dev" "out" "man" ] ++ lib.optional pythonSupport "py";
 
   patches = stdenv.lib.optional stdenv.hostPlatform.isMusl
     (fetchpatch {
@@ -21,7 +22,22 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ autoreconfHook bison flex pkgconfig ];
+  nativeBuildInputs = [ autoreconfHook bison flex pkgconfig file ]
+    ++ lib.optional pythonSupport swig;
+
+  postBuild = lib.optionalString (pythonSupport) ''
+      cd python
+      ${python}/bin/python setup.py install --prefix=../pythonlib
+      cd -
+  '';
+
+  postFixup = lib.optionalString pythonSupport ''
+    mv "pythonlib/" "$py"
+  '';
+
+  passthru = {
+    inherit pythonSupport;
+  };
 
   meta = with lib; {
     inherit version;

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -6951,6 +6951,9 @@ in {
 
   pylru = callPackage ../development/python-modules/pylru { };
 
+  libnl-python = disabledIf isPy3k
+    (toPythonModule (pkgs.libnl.override{pythonSupport=true; inherit python; })).py;
+
   lark-parser = callPackage ../development/python-modules/lark-parser { };
 
   lazy-object-proxy = buildPythonPackage rec {

